<a href="https://colab.research.google.com/github/manisankar29/Deep-Learning/blob/main/CNN_classification.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **CNN (Convolutional Neural Network) Classification**

## Import necessary libraries

In [1]:
from tensorflow.keras.datasets import mnist
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Conv2D, MaxPool2D, Flatten, Dropout, Dense

## Import and Extract 'mnist' dataset

In [2]:
(X_train, Y_train),(X_test, Y_test) = mnist.load_data()

11490434/11490434 [==============================] - 0s 0us/step


In [3]:
X_train = X_train.reshape((X_train.shape[0], X_train.shape[1], X_train.shape[2], 1))
X_test = X_test.reshape((X_test.shape[0], X_test.shape[1], X_test.shape[2], 1))

In [4]:
X_train.shape

(60000, 28, 28, 1)

In [5]:
X_test.shape

(10000, 28, 28, 1)

In [6]:
X_train = X_train / 255
X_test = X_test / 255

## Model Development and Evaluation

In [9]:
model = Sequential()
model.add(Conv2D(32, (3,3), activation='relu', input_shape=(28,28,1)))
model.add(MaxPool2D(2,2))
model.add(Flatten())
model.add(Dense(100, activation='relu'))
model.add(Dense(10, activation='softmax'))
model.compile(loss='sparse_categorical_crossentropy', optimizer='adam', metrics=['accuracy'])
model.fit(X_train, Y_train, epochs=10)
model.evaluate(X_test, Y_test)

Epoch 1/10
1875/1875 [==============================] - 9s 3ms/step - loss: 0.1642 - accuracy: 0.9509
Epoch 2/10
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0557 - accuracy: 0.9827
Epoch 3/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.0368 - accuracy: 0.9886
Epoch 4/10
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0252 - accuracy: 0.9919
Epoch 5/10
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0175 - accuracy: 0.9942
Epoch 6/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.0130 - accuracy: 0.9959
Epoch 7/10
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0091 - accuracy: 0.9971
Epoch 8/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.0076 - accuracy: 0.9974
Epoch 9/10
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0060 - accuracy: 0.9982
Epoch 10/10
313/313 [==============================] - 1s 3ms/step - loss: 0.0522 

[0.05219797417521477, 0.9858999848365784]

The final accuracy achieved by CNN classification model is 98.59%.